parsers/rebase/rebase.go

```diff
@@ -0,0 +1,264 @@
+}
+
+// ParseRebase parses the Rebase database into a map of enzymes
+func ParseRebase(file []byte) map[string]Enzyme {
+	// Setup some variables
+	var enzyme Enzyme
+	enzymeMap := make(map[string]Enzyme)
+	commercialSupplierMap := make(map[rune]string)
+
+	// Get rebase as a large string
+	rebase := string(file)
+
+	// Split those strings into individual lines for parsing
+	lines := strings.Split(rebase, "\n")
+
+	commercialParsingLine := 0
+	startCommercialParsing := false
+	for _, line := range lines {
+		// Parse commercial sources map
+		if line == "REBASE codes for commercial sources of enzymes" {
+			startCommercialParsing = true
+		}
+		// If startCommercialParsing is true, start building the commercial supplier map
+		if startCommercialParsing {
+			// if we start enzyme parsing, break the commercial supplier parsing
+			if strings.Contains(line, "<1>") {
+				commercialParsingLine = 0
+				startCommercialParsing = false
+			}
+
+			// Skip two lines
+			commercialParsingLine++
+			if (commercialParsingLine > 3) && (len(strings.TrimLeft(line, "\t")) > 0) {
+				// Trim indentation
+				trimmedString := strings.TrimLeft(line, "\t")
+
+				// The first letter of the trimmedString is the single letter code
+				singleLetterCommercialCode := rune(trimmedString[0])
+
+				// There are 8 spaces until the commercial companies's name. We are keeping the dates
+				// attached, since it is additional information that could be useful for users down
+				// the line
+				commercialName := trimmedString[9:]
+
+				// Add both to commercialSuppliermap
+				commercialSupplierMap[singleLetterCommercialCode] = commercialName
+			}
+		}
+
+		// Normal enzyme parsing
+		switch {
+		case strings.Contains(line, "<1>"):
+			enzyme.Name = line[3:]
+		case strings.Contains(line, "<2>"):
+			enzyme.Isoschizomers = strings.Split(line[3:], ",")
+		case strings.Contains(line, "<3>"):
+			enzyme.RecognitionSequence = line[3:]
+		case strings.Contains(line, "<4>"):
+			enzyme.MethylationSite = line[3:]
+		case strings.Contains(line, "<5>"):
+			enzyme.MicroOrganism = line[3:]
+		case strings.Contains(line, "<6>"):
+			enzyme.Source = line[3:]
+		case strings.Contains(line, "<7>"):
+			// We need to get a list of specific commercial suppliers from the commercialSupplierMap we previously made
+			var commercialSuppliers []string
+			for _, commercialLetter := range line[3:] {
+				commercialSuppliers = append(commercialSuppliers, commercialSupplierMap[commercialLetter])
+			}
+			enzyme.CommercialAvailability = commercialSuppliers
+		case strings.Contains(line, "<8>"):
+			enzyme.References = line[3:]
+			// After every <8> a new enzyme will start. So here, we put the current enzyme into the enzymeMap and setup a new enzyme to be filled
+			enzymeMap[enzyme.Name] = enzyme
+			enzyme = Enzyme{}
+		}
+	}
+	return enzymeMap
+}
+
+// ReadRebase returns an enzymeMap from a Rebase data dump
+func ReadRebase(path string) (map[string]Enzyme, error) {
```

Comment:
`ReadRebase` makes sense to developers but should we use a name like `GetEnzymeMap` for those who don't know what rebase is?

Reply:
Should we have a function GetEnzymeMap that downloads the file directly from NEB and parses? Seems like we'd be hitting their endpoints quite hard in that case.

Reply:
Since it would run on every test.